gk/main.c

```diff
@@ -1412,6 +1417,281 @@ update_ip_hop_count(struct gatekeeper_if *iface, struct ipacket *packet,
 	return 0;
 }
 
+static void
+lookup_fib_bulk(struct gk_lpm *ltbl, struct ip_flow **flows,
+	int num_flows, struct gk_fib *fibs[])
+{
+	int i;
+	int k = RTE_ALIGN_FLOOR(num_flows, FWDSTEP);
+
+	for (i = 0; i != k; i += FWDSTEP) {
+		int j;
+		__m128i dip = _mm_set_epi32(flows[i + 3]->f.v4.dst.s_addr,
+			flows[i + 2]->f.v4.dst.s_addr,
+			flows[i + 1]->f.v4.dst.s_addr,
+			flows[i]->f.v4.dst.s_addr);
+		rte_xmm_t dst;
+		uint16_t dprt[FWDSTEP];
+
+		rte_lpm_lookupx4(ltbl->lpm, dip, dst.u32, 65535);
```

Comment:
Define a `const` in the code instead of repeating `65535`.

Reply:
1. Aren't we assuming elsewhere that the next hop field is more than 16 bits long? Isn't it possible to assume it's 32 bits long here?

2. Whatever value we use as the last parameter of `rte_lpm_lookupx4()`, if it's just a bunch of ones like 65535, it would be more readable to write in hexadecimal (e.g. `0xFFFF` for 65535).

Reply:
1. You are right, the latest version uses 24 bits long next hop, as shown in [DPDK LPM library](https://github.com/cjdoucette/dpdk/blob/ec313047f6b784f61739331f8383ac658a92eb6a/lib/librte_lpm/rte_lpm.h#L127-L133).